plugins/mac-syphon/CMakeLists.txt

```diff
@@ -11,76 +11,19 @@ else()
-          syphon-framework/SyphonCFMessageReceiver.m
-          syphon-framework/SyphonCFMessageReceiver.h
-          syphon-framework/SyphonCFMessageSender.h
-          syphon-framework/SyphonCFMessageSender.m
-          syphon-framework/SyphonClient.m
-          syphon-framework/SyphonClient.h
-          syphon-framework/SyphonClientConnectionManager.m
-          syphon-framework/SyphonClientConnectionManager.h
-          syphon-framework/SyphonDispatch.c
-          syphon-framework/SyphonDispatch.h
-          syphon-framework/SyphonIOSurfaceImage.m
-          syphon-framework/SyphonIOSurfaceImage.h
-          syphon-framework/SyphonImage.m
-          syphon-framework/SyphonImage.h
-          syphon-framework/SyphonMachMessageReceiver.m
-          syphon-framework/SyphonMachMessageReceiver.h
-          syphon-framework/SyphonMachMessageSender.m
-          syphon-framework/SyphonMachMessageSender.h
-          syphon-framework/SyphonMessageQueue.m
-          syphon-framework/SyphonMessageQueue.h
-          syphon-framework/SyphonMessageReceiver.m
-          syphon-framework/SyphonMessageReceiver.h
-          syphon-framework/SyphonMessageSender.m
-          syphon-framework/SyphonMessageSender.h
-          syphon-framework/SyphonMessaging.m
-          syphon-framework/SyphonMessaging.h
-          syphon-framework/SyphonOpenGLFunctions.c
-          syphon-framework/SyphonOpenGLFunctions.h
-          syphon-framework/SyphonPrivate.m
-          syphon-framework/SyphonPrivate.h
-          syphon-framework/SyphonServer.m
-          syphon-framework/SyphonServer.h
-          syphon-framework/SyphonServerConnectionManager.m
-          syphon-framework/SyphonServerConnectionManager.h
-          syphon-framework/SyphonServerDirectory.m
-          syphon-framework/SyphonServerDirectory.h)
-
-target_compile_options(
-  syphon-framework
-  PRIVATE -include "${CMAKE_CURRENT_SOURCE_DIR}/syphon-framework/Syphon_Prefix.pch" -Wno-error=deprecated-declarations
-  PUBLIC -Wno-error=newline-eof -Wno-error=deprecated-implementations)
-
-target_compile_definitions(syphon-framework PUBLIC "SYPHON_UNIQUE_CLASS_NAME_PREFIX=OBS_")
-
-target_link_libraries(syphon-framework PUBLIC OpenGL::GL ${COCOA} ${IOSURF})
-
-set_target_properties(syphon-framework PROPERTIES FOLDER "plugins/mac-syphon" PREFIX "")
+find_library(SYPHON Syphon)
+mark_as_advanced(APPKIT IOSURFACE SCRIPTINGBRIDGE SYPHON)
 
 add_library(mac-syphon MODULE)
 add_library(OBS::syphon ALIAS mac-syphon)
 
 target_sources(mac-syphon PRIVATE syphon.m plugin-main.c)
 
-target_compile_options(mac-syphon PRIVATE -include "${CMAKE_CURRENT_SOURCE_DIR}/syphon-framework/Syphon_Prefix.pch"
-                                          -fobjc-arc)
-
-target_link_libraries(mac-syphon PRIVATE OBS::libobs Syphon::Framework ${SCRIPTINGBRIDGE})
+target_compile_options(mac-syphon PRIVATE -fobjc-arc)
+target_link_libraries(mac-syphon PRIVATE OBS::libobs ${APPKIT} ${IOSURFACE} ${SCRIPTINGBRIDGE} ${SYPHON})
```

Comment:
If we're changing these parts here, we might as well clean them up correctly:

Instead of `find_library` and a CMake variable, use `$<LINK_LIBRARY:Framework,AppKit.framework>` to link against system frameworks and with `${SYPHON}` to make it explicit.

Reply:
Changed to use `LINK_LIBRARY:FRAMEWORK` for all frameworks.